beprof/curve.py

```diff
@@ -113,7 +113,8 @@ def change_domain(self, domain):
             raise ValueError('in change_domain():' 'the old domain does not include the new one')
 
         y = np.interp(domain, self.x, self.y)
-        obj = self.__class__(np.stack((domain, y), axis=1), **self.__dict__['metadata'])
+        # np.dstack(...)[0] is used to extract nested array (previously used np.stack which behaved different)
+        obj = self.__class__(np.dstack((domain, y))[0], **self.__dict__['metadata'])
```

Comment:
Not really, I had in mind a description for a fellow programmer who takes over the project and is trying to figure out what is going on in these few lines of code, sth like:

```We need to ... domain and y because .... We pass it as argument to ... because it takes .... Dstast is ... and this is the reason why we use [0]```